packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1824,9 +1824,90 @@ function scheduleWorkToRoot(fiber: Fiber, expirationTime): FiberRoot | null {
   return root;
 }
 
-export function warnIfNotCurrentlyBatchingInDev(fiber: Fiber): void {
+// in a test-like environment, we want to warn if dispatchAction() is
+// called outside of a TestUtils.act(...)/batchedUpdates/render call.
+// so we have a a step counter for when we descend/ascend from
+// actedUpdates() calls, and test on it for when to warn
+let actingUpdatesScopeDepth = 0;
+
+export function actedUpdates(
+  callback: () => void | Promise<void>,
+  runCallbackUntilPredicateFails: (
+    callback: () => void,
+    predicate: () => boolean,
```

Comment:
`runCallbackUntilPredicateFails` isn't a public API, right? Maybe I missed a conversation about this?

Reply:
I had expected you'd solve this problem with something like `doesHavePendingPassiveEffects`. Why is this preferable? Remember this is a temporary workaround and the real solution is to have a custom testing build of React DOM.

Reply:
Have a look now? Do you think I should still export `doesHavePassiveEffects` instead of passing it as a callback?